### PR TITLE
Create `update-version` script to more easily update version

### PR DESCRIPTION
**Pull request by spikebrehm**
Similar to `npm version`, but updates the version number in all the
files we need to.

Usage:

    ./scripts/update-version.js [major|minor|patch]

It will:

* Get current npm version.
* Determine what type of version bump: major, minor, patch
* Update the files:
  * package.json
  * android/gradle.properties
  * react-native-maps.podspec
  * react-native-google-maps.podspec
* Make Git commit.
* Create Git tag.

to: @lelandrichardson @gilbox 
cc: @ljharb

### CODE REVIEW

**Comment by ljharb** [comment]
If you add this script under "preversion", "version", or "postversion" then `npm version` should invoke it for you.

File: scripts/update-version.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+// eslint-disable-next-line import/no-extraneous-dependencies
+const semver = require('semver');
+const pkg = require('../package.json');
+
+const versionTypes = ['major', 'minor', 'patch'];
+const filesToUpdate = [
+  'package.json',
+  'android/gradle.properties',
+  'react-native-maps.podspec',
+  'react-native-google-maps.podspec',
```
Comment (ljharb) [inline scripts/update-version.js]: are all three of these always in sync? What happens when there's a bugfix that only applies to one of them?

**Comment by gilbox** [comment]
very cool! 👍

**Comment by spikebrehm** [comment]
ok PTAL Jordan

File: package.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
   "version": "0.13.0",
   "scripts": {
     "start": "react-native start",
-    "lint": "eslint ."
+    "lint": "eslint .",
+    "preversion": "./scripts/update-version.js"
```
Comment (ljharb) [inline package.json]: the `.js` isn't necessary since it has a shebang, fwiw

**Pull request merged**